src/util/hash.rs

```diff
@@ -64,7 +64,7 @@ pub fn bitcoin_merkle_root<T, I>(mut iter: I) -> T
         return Default::default();
```

Comment:
Now I noticed this compares with `0` but above it calls `is_empty`. Would be nice to make those same too.
But what worries me more is returning all-zeroes hash if the input is empty. It doesn't seem to make sense. I think it should either panic or return `Option<T>`.